File: SQL/Select/SQLSelectTableFromSelect.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLSelectTableFromSelect
        Inherits SQLSelectTableBase

        Private pobjSelect As SQLSelect

        Public Sub New(ByVal objSelect As SQLSelect, ByVal strAlias As String)

            If strAlias = String.Empty Then
                Throw New ArgumentNullException
            ElseIf objSelect Is Nothing Then
                Throw New ArgumentNullException
            End If

            pobjSelect = objSelect
            MyBase.Alias = strAlias

        End Sub

        Friend Overrides Function GetPrefix() As String

            Return MyBase.Alias

        End Function
         
        Protected Overrides ReadOnly Property Source(ByVal eConnectionType As Database.ConnectionType) As String
            Get

                pobjSelect.ConnectionType = eConnectionType
                Return "(" & pobjSelect.SQL & ")"

            End Get
        End Property

    End Class

End Namespace

File: SQL/Select/SQLSelectTableFromSelect.cs
// ___________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLSelectTableFromSelect : SQLSelectTableBase
	{
		private SQLSelect pobjSelect;
			
		public SQLSelectTableFromSelect(SQLSelect objSelect, string strAlias)
		{
			if (String.IsNullOrEmpty(strAlias))
				throw new ArgumentNullException();
			else if (objSelect == null)
				throw new ArgumentNullException();
				
			pobjSelect = objSelect;
			base.Alias = strAlias;
		}

		public SQLSelect Select
		{
			get
			{
				return pobjSelect;
			}
		}
					
		internal override string Source(Serializers.Serializer serializer)
		{
			return serializer.SerializeSelectTableFromSelect(this);
		}
	}
}
